src/main/java/com/epam/reportportal/junit/ReportPortalListener.java

```diff
@@ -1,104 +1,95 @@
-/*
- * Copyright 2016 EPAM Systems
- *
- *
- * This file is part of EPAM Report Portal.
- * https://github.com/reportportal/agent-java-junit/
- *
- * Report Portal is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * Report Portal is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
- */
-package com.epam.reportportal.junit;
-
-import com.epam.reportportal.listeners.Statuses;
-import org.junit.runner.Description;
-import org.junit.runner.Result;
-import org.junit.runner.notification.Failure;
-import org.junit.runner.notification.RunListener;
-
-/**
- * Report portal custom event listener. This listener support parallel running
- * of tests and test methods. Main constraint: All test classes in current
- * launch should be unique. (User shouldn't run the same classes twice/or more
- * times in the one launch)
- *
- * @author Aliaksei_Makayed (modified by Andrei_Ramanchuk)
- */
-
-public class ReportPortalListener extends RunListener {
-
-	private IListenerHandler handler = JUnitInjectorProvider.getInstance().getBean(IListenerHandler.class);
-
-	/**
-	 * {@inheritDoc}
-	 */
-	@Override
-	public void testRunStarted(Description description) throws Exception {
-		handler.startLaunch();
-		handler.initSuiteProcessor(description);
-	}
-
-	/**
-	 * {@inheritDoc}
-	 */
-	@Override
-	public void testStarted(Description description) throws Exception {
-		handler.startSuiteIfRequired(description);
-		handler.starTestIfRequired(description);
-		handler.startTestMethod(description);
-
-	}
-
-	/**
-	 * {@inheritDoc}
-	 */
-	@Override
-	public void testFinished(Description description) throws Exception {
-		// <Andrei Varabyeu>: it isn't completely why validation for # symbol presence has been added here:
```

Comment:
The original code that checks for the '#' character appears to relate to the CustomJUnitRunner class, which alters the names of configuration methods to distinguish then from those of test methods. However, this code appears never to have been finished. As a result, lifecycle events for ordinary test methods never got posted to the ReportPortal backend.